src/lib/plugin.js

```diff
@@ -248,62 +249,80 @@ module.exports = class PluginPaymentChannel extends EventEmitter2 {
     this._safeEmit('outgoing_request', message)
 
     this.debug('requesting with plugin', message)
-    const btpResponse = await this._rpc.message(ilpAndCustomToProtocolData(message))
-
-    const { ilp, custom } = protocolDataToIlpAndCustom(btpResponse)
+    let protocolMapOut = Object.assign({}, message.custom, { ilp: message.ilp })
+    const btpResponse = await this._rpc.message(protocolMapToProtocolData(protocolMapOut))
+    let protocolMapIn = protocolDataToProtocolMap(btpResponse.protocolData)
     const parsed = {
       to: this.getAccount(),
       from: this.getPeerAccount(),
       ledger: this._prefix
     }
+    if (protocolMapIn.ilp) {
+      parsed.ilp = protocolMapIn.ilp
+      delete protocolMapIn.ilp
+    }
 
-    if (ilp) parsed.ilp = ilp
-    if (custom) parsed.custom = custom
+    if (Object.keys(protocolMapIn).length) {
+      parsed.custom = protocolMapIn
+    }
 
     this._validator.validateIncomingMessage(parsed)
     this._safeEmit('incoming_response', parsed)
-
     return parsed
   }
 
   async _handleRequest ({requestId, data}) {
-    const { ilp, custom, protocolMap } = protocolDataToIlpAndCustom(data)
+    let protocolMapIn = protocolDataToProtocolMap(data.protocolData)
     const message = {
       id: requestId,
       to: this.getAccount(),
       from: this.getPeerAccount()
     }
-
-    if (ilp) message.ilp = ilp
-    if (custom) message.custom = custom
-
-    // if there are side protocols only
-    if (!ilp) {
-      if (protocolMap.get_info) {
-        return [{
-          protocolName: 'get_info',
-          contentType: Btp.MIME_APPLICATION_JSON,
-          data: Buffer.from(JSON.stringify(this.getInfo()))
-        }]
-      } else if (protocolMap.get_balance) {
+    if (protocolMapIn.ilp) {
+      message.ilp = protocolMapIn.ilp
+      delete protocolMapIn.ilp
+    }
+    message.custom = protocolMapIn
+    // special requests that do not trigger the registered request handler
+    switch (protocolMapIn.primary) {
+      case 'info':
+        if (typeof protocolMapIn.info.data !== 'string') {
+          throw new Error('info requests should come as base64url-encoded octet stream data')
+        }
+        const requestBuffer = Buffer.from(protocolMapIn.info.data, 'base64')
```

Comment:
I recently asked myself that same question about `Transfer#ilp` and `Message#ilp` in the LPI.

Doing it here not just for the ilp protocolData entry but also for other binary protocolData entries is mainly for symmetry.

We can't change the LPI, but I'm open to making it so that although `Message#ilp` is base64url-encoded, Message#custom.vouch is a Buffer object - that would not violate the LPI.